setup/express.js

```diff
@@ -0,0 +1,122 @@
+const expressWinston = require('express-winston');
+
+const { setupNodeLogger } = require('./winston');
+const { DEFAULT_LOG_LEVEL } = require('..');
+
+const defaultFilters = [
+  { agent: 'Consul Health Check', urls: ['/'], statuses: [200] },
+  { agent: 'Pingdom.com_bot_version', urls: ['/', '/api/v1/health'], statuses: [200] },
+];
+
+function skipRequests(filters) {
+  return (req, res) => {
+    const userAgent = req.headers['user-agent'] || '';
+
+    return filters.some(
+      ({ agent, urls, statuses }) => userAgent.includes(agent)
+        && urls.includes(req.url)
+        && statuses.includes(res.statusCode),
+    );
+  };
+}
+
+const whitelistedHeaders = [
+  'x-forwarded-for',
+  'x-forwarded-proto',
+  'content-type',
+];
+
+function ecsMeta(req, res) {
+  const meta = {};
+  meta.http = {};
+  meta.http.protocol = req.httpVersion;
+
+  meta.http.request = {};
+  meta.http.request.method = req.method.toLowerCase();
+  meta.http.request.bytes = req.socket.bytesRead;
+
+  if (req.headers.referer) {
+    meta.http.request.referer = req.headers.referer;
+  }
+
+  if (req.headers['content-length']) {
+    meta.http.request.body = {};
+    meta.http.request.body.bytes = req.headers['content-length'];
+  }
+
+  if (req.headers['user-agent']) {
+    meta.user_agent = {};
+    meta.user_agent.original = req.headers['user-agent'];
+  }
+
+  whitelistedHeaders.forEach((header) => {
+    if (req.headers[header]) {
+      meta.http.request.headers = meta.http.request.headers || {};
+      meta.http.request.headers[header] = req.headers[header];
+    }
+  });
+
+  meta.url = {};
+  meta.url.original = req.originalUrl;
+  meta.url.scheme = req.protocol;
+  if (req.hostname) {
+    const [host, port] = req.hostname.split(':');
+    meta.url.domain = host;
+    if (port) meta.url.port = Number(port);
+  }
+
+  meta.client = {};
+  meta.client.ip = req.ip;
+  meta.client.port = req.socket.remotePort;
+
+  meta.server = {};
+  meta.server.ip = req.socket.localAddress;
+  meta.server.port = req.socket.localPort;
+
+  if (res) {
+    meta.http = meta.http || {};
+    meta.http.response = {};
+    meta.http.response.status_code = res.statusCode;
+    meta.event = {};
+    meta.event.duration_ms = res.responseTime;
+  }
+
+  return meta;
```

Comment:
Not critical, but stylistic: wouldn't it be a lot cleaner to just create the object on the fly? Or even create optional overrides and spread them in?

Reply:
can you give an example?

Reply:
Something like:
```
const userAgent = req.headers['user-agent'] ? { user_agent: { original: req.headers['user-agent'] } } : {};   // Maybe can clean it up too

return {
  http: {
    protocol: req.httpRevision,
    // more stuff here
  },
  // more stuff here
  ... userAgent,
}; 
```

Reply:
mmm, interesting. I can try and play around with it. But I do want to avoid empty dicts being in the final output.

Reply:
changed it, let me know if it's any better